### DISCUSSION TITLE
Add a type of log file that keeps track of all the downloaded beatmap sets by ID in order to skip already downloaed maps when downloading in bulk

### DISCUSSION THREAD

**Discussion by IVII1** [ppy/osu#25310] [Ideas] [answered]
Make it so each time new beatmaps are imported, this file gets referenced in order to skip maps that are already in the log file. This file can later be used by [osu!Collector](https://osucollector.com/) and similar third party websites to skip downloads of maps that are already in the log file. Delete all beatmaps when importing regardless of weather they're new or not to save space.

Can this be optimized? I understand that it may be a lengthy process especially if someone has 10k+ maps but this would help me since i like to bulk download maps once i get bored of ones i already have downloaded. 

Let me know about if the technical side of this is plausible from the technical side.

**Comment by peppy** [comment, marked as answer]
Sounds out of scope. You can pull this info from the realm database if you need to.
